base/Dockerfile

```diff
@@ -1,22 +1,31 @@
-FROM ubuntu:xenial
+FROM python:2-slim
 
 EXPOSE 5000
 
 RUN useradd --create-home redash
 
 # Ubuntu packages
-RUN apt-get update && apt-get install -y curl && curl https://deb.nodesource.com/setup_6.x | bash - && \
-  apt-get install -y python-pip python-dev build-essential pwgen libffi-dev sudo git-core wget \
-  nodejs \
-  # Postgres client
-  libpq-dev \
-  # for SAML
-  xmlsec1 \
-  # Additional packages required for data sources:
-  libssl-dev libmysqlclient-dev freetds-dev libsasl2-dev && \
+RUN apt-get update && \
+  apt-get install -y curl gnupg && \
+  curl https://deb.nodesource.com/setup_6.x | bash - && \
```

Comment:
Interesting, couldn't we make this base image install both Python and Node.js as well?

Reply:
We can, but why have this extra baggage?

Only reason I can think of

Reply:
We can, but why have this extra baggage?

Only reason I can think of is if we want to move the webpack build process in development to happen in a Docker container instead of on the host. But maybe for this case we should use a different Docker file.

Here's how the multi from Dockerfile looks like:

```Dockerfile
FROM node:8.12 AS frontend
WORKDIR /app
COPY package.json  package-lock.json ./
RUN npm install
COPY . ./
RUN npm run build

FROM redash/base:debian

# Controls whether to install extra dependencies needed for all data sources.
ARG skip_ds_deps

# We first copy only the requirements file, to avoid rebuilding on every file
# change.
COPY requirements.txt requirements_dev.txt requirements_all_ds.txt ./
RUN pip install -r requirements.txt -r requirements_dev.txt
RUN if [ "x$skip_ds_deps" = "x" ] ; then pip install -r requirements_all_ds.txt ; else echo "Skipping pip install -r requirements_all_ds.txt" ; fi

COPY . ./
COPY --from=frontend /app/client /app/client
RUN chown -R redash /app
USER redash

ENTRYPOINT ["/app/bin/docker-entrypoint"]
CMD ["server"]
```

Reply:
Ah, you're right, I forgot we don't need Node once the image with the bundle is built. I like that!